### PR TITLE
[chore]: cleanup and refactoring of component generator

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by saurabh100ni** [comment]
@VihasMakwana If there is anything you want to discuss about this PR, Please add this as your agenda to [Docs](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit) for the Meshery Development Meet which is scheduled at 8:30 PM IST today.

**Comment by VihasMakwana** [comment]
> @VihasMakwana If there is anything you want to discuss about this PR, Please add this as your agenda to [Docs](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit) for the Meshery Development Meet which is scheduled at 8:30 PM IST today.

This is in progress, I would like to discuss it later when it's up for review.

**Comment by saurabh100ni** [comment]
okay, sure.

**Comment by leecalcote** [comment]
Ready for review.

File: scripts/component_generation/main.go [hunk 1]
```diff
@@ -17,15 +15,28 @@ import (
...
 )
 
 const dumpFile = "./dump.csv"
 const COLUMNRANGE = "!A:AH3" //Update this on addition of new columns
 const APPENDRANGE = "!A4:V4"
+const sheetID = 0
+
+var spreadsheetID = "1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdwizOJmeMw"
+
+type componentWrapper struct {
```
Comment (leecalcote) [inline scripts/component_generation/main.go:28]: @MUzairS15 @Yashs911 am I make sense that this point to Meshery’s preferred Helmrepo connection provider: the model for Flux.

File: scripts/component_generation/main.go [hunk 2]
```diff
@@ -130,25 +138,16 @@ func main() {
 			fmt.Println(err)
 			return
 		}
-		err = writeComponentModels(pkgs, &modelsWriter)
+		err = writeComponentModels(pkgs, modelsFd)
```
Comment (leecalcote) [inline scripts/component_generation/main.go:141]: We can begin moving these errors over to MeshKit-style errors

File: scripts/component_generation/main.go [hunk 3]
```diff
@@ -275,149 +281,50 @@ func (d *dedup) set(key string) {
...
+	for pkg := range pkgChan {
+		if err := pkg.UpdatePackageData(); err != nil {
+			fmt.Println(err)
+			continue
 		}
-	}()
-	if _, err := os.Stat(dumpFile); os.IsExist(err) {
-		// If file exists, delete it
-		err := os.Remove(dumpFile)
+		fmt.Printf("[DEBUG] Generating components for: %s with verified status %v\n", pkg.Name, pkg.VerifiedPublisher)
```
Comment (MUzairS15) [inline scripts/component_generation/main.go:291]: Can we use `mehskit logger`?

File: scripts/component_generation/spreadsheet.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	availableComponentsPerModel map[string]map[string]bool
+}
+
+func NewSheetSRV() *sheets.Service {
+	ctx := context.Background()
+	byt, _ := base64.StdEncoding.DecodeString(os.Getenv("CRED"))
+	// authenticate and get configuration
+	config, err := google.JWTConfigFromJSON(byt, "https://www.googleapis.com/auth/spreadsheets")
+	if err != nil {
+		fmt.Println("ERR2", err)
```
Comment (MUzairS15) [inline scripts/component_generation/spreadsheet.go:28]: As you refactor please ensure to have meaningful error msgs and logs.

File: scripts/component_generation/spreadsheet.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+		return nil
+	}
+	// create client with config and context
+	client := config.Client(ctx)
+	// create new service using client
+	srv, err := sheets.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		fmt.Println("ERR3", err)
```
Comment (MUzairS15) [inline scripts/component_generation/spreadsheet.go:36]: Same here.

**Comment by VihasMakwana** [comment]
@leecalcote @MUzairS15 I've moved to meshkit logging and added some ErrorCodes. PTAL.

**Comment by leecalcote** [comment]
…and you’re using the MeshKit logger as well. 🎯

**Comment by leecalcote** [comment]
@MUzairS15 a penny for your thoughts...

**Comment by leecalcote** [comment]
> @MUzairS15 a penny for your thoughts...

Ok, two pennies for your thoughts, @MUzairS15

**Review by MUzairS15 — COMMENTED** [review]
Decided an alternative approach.

**Pull request closed without merging**